Make needed updates to run forecast from GEFS

XianwuXue-NOAA opened the pull request:
**Description**

Since some of the GEFS functions are empty or missing, this PR is to complete/add such functions and also make some variables to accept values from other scripts

Refs: #1147

**Type of change**

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**

Tested on WCOSS2 using GEFS workflow
  
**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing tests pass with my changes

XianwuXue-NOAA committed:
Improve to run fcst (atm only)

 On branch feature/gefs_v13_822_add_atm_fcst
	modified:   jobs/rocoto/fcst.sh
	modified:   ush/forecast_det.sh
	modified:   ush/forecast_postdet.sh
	modified:   ush/forecast_predet.sh

Refs: #822

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/gefs_v13_822_add_atm_fcst

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/gefs_v13_822_add_atm_fcst

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/gefs_v13_822_add_atm_fcst

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/gefs_v13_822_add_atm_fcst

XianwuXue-NOAA committed:
Improve the script to run fcst for GEFS

 On branch feature/gefs_v13_822_add_atm_fcst
	modified:   jobs/JGLOBAL_FORECAST
	modified:   jobs/rocoto/fcst.sh
	modified:   parm/config/config.fcst
	modified:   ush/forecast_postdet.sh
	modified:   ush/forecast_predet.sh

Refs: #822

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/gefs_v13_822_add_atm_fcst

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/gefs_v13_822_add_atm_fcst

 Conflicts:
	ush/forecast_predet.sh

Refs: #822

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/gefs_v13_822_add_atm_fcst

XianwuXue-NOAA committed:
Modify to run atm fcst for GEFS

 On branch feature/gefs_v13_822_add_atm_fcst
 Changes to be committed:
	modified:   ush/forecast_postdet.sh
	modified:   ush/forecast_predet.sh

Refs: #822

XianwuXue-NOAA committed:
Fix bugs for condition statements

 On branch feature/gefs_v13_822_add_atm_fcst
 Changes to be committed:
	modified:   ush/forecast_predet.sh

Refs: #822

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/gefs_v13_822_add_atm_fcst

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/gefs_v13_822_add_atm_fcst

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/gefs_v13_822_add_atm_fcst

XianwuXue-NOAA committed:
Introduce FHOUT_ENKF_GFS for gfsefcs jobs
    and increase alltime of efcs jobs

 On branch feature/gefs_v13_822_add_atm_fcst
	modified:   parm/config/config.base.emc.dyn
	modified:   parm/config/config.resources
	modified:   scripts/exgdas_enkf_fcst.sh

Refs: #822

XianwuXue-NOAA committed:
Add FHOUT_ENKF_GFS to calculate epos group

 On branch feature/gefs_v13_822_add_atm_fcst
 Changes to be committed:
	modified:   workflow/rocoto/workflow_tasks.py

Refs: #822

XianwuXue-NOAA committed:
Add GEFS functions with new COM to fcst ush scripts
 On branch feature/gefs_v13_822_add_atm_fcst
 Changes to be committed:
	modified:   ush/forecast_det.sh
	modified:   ush/forecast_postdet.sh
	modified:   ush/forecast_predet.sh

Refs: #822

XianwuXue-NOAA committed:
Remove unused statements for gefs

 On branch feature/gefs_v13_822_add_atm_fcst
 Changes to be committed:
	modified:   parm/config/config.fcst

Refs: #822

XianwuXue-NOAA committed:
Add FLTFILEGFS[F00] & POSTGRB2TBL to pass values

from external packages such as GEFS

 On branch feature/gefs_v13_822_add_atm_fcst
	modified:   ush/forecast_postdet.sh

Refs: #822

WalterKolczynski-NOAA commented on parm/config/config.resources:
```diff
@@ -737,7 +737,7 @@ elif [ ${step} = "esfc" ]; then
 
     if [ ${CASE} = "C768" ]; then
-      export wtime_efcs="01:20:00"
+      export wtime_efcs="06:00:00"
```
This is a big change, and only for one resolution. Seems like there should be a `wtime_efcs_gfs` (and possibly other (`_gfs` settings) now to cover early cycle?

XianwuXue-NOAA replied:
In fact, using "01:20:00" is only good for coarse resolution, if we want to run C768/C384, then the walltime is too short. Therefore, I increase the wall time. I agree that we may have an another variable to cover early cycle. However, in the early-enkf, we may not run gfsemn jobs since it will be run GEFSv13.

WalterKolczynski-NOAA commented on ush/forecast_postdet.sh:
```diff
@@ -519,11 +520,19 @@ EOF
-      logo=$memdir/${CDUMP}.t${cyc}z.logf${FH3}.txt
-      pgbo=$memdir/${CDUMP}.t${cyc}z.master.grb2f${FH3}
-      flxo=$memdir/${CDUMP}.t${cyc}z.sfluxgrbf${FH3}.grib2
+      if [[ $RUN == "gefs" ]]; then
```
This should be temporary until GFS is updated to new COM structure.

XianwuXue-NOAA replied:
Yes.

WalterKolczynski-NOAA replied:
Maybe we should just make a `$HISTORY_DIR` and a `$MASTER_DIR` and set appropriately instead of all these if-blocks.

XianwuXue-NOAA replied:
I agree, however, it will add more variables. If you agree, I can do this changes.

WalterKolczynski-NOAA commented on ush/forecast_postdet.sh:
```diff
@@ -534,11 +543,19 @@ EOF
-      eval $NLN grid_spec.tile${n}.nc     $memdir/grid_spec.tile${n}.nc
-      eval $NLN atmos_static.tile${n}.nc  $memdir/atmos_static.tile${n}.nc
-      eval $NLN atmos_4xdaily.tile${n}.nc $memdir/atmos_4xdaily.tile${n}.nc
+      if [[ $RUN == "gefs" ]]; then
```
This too

XianwuXue-NOAA replied:
yes

WalterKolczynski-NOAA commented on ush/forecast_postdet.sh:
```diff
@@ -556,6 +573,20 @@ FV3_GFS_nml(){
   echo SUB ${FUNCNAME[0]}: FV3 name lists and model configure file created
 }
 
+FV3_GEFS_nml(){
```
This is identical to FV3_GFS_nml, why is it needed?

WalterKolczynski-NOAA replied:
There are also a bunch of "GEFS" functions added that just call the GFS version. Is there a need to do that?

XianwuXue-NOAA replied:
I added these functions is based on the exglobal_forecast.sh. Such as: Since the function should be similar with GFS functions, I just reuse the GFS functions. 

case $RUN in
  'data') DATM_nml;;
  'gfs') FV3_GFS_nml;;
  'gdas') FV3_GFS_nml;;
  'gefs') FV3_GEFS_nml;;
esac				#no namelist for data atmosphere

WalterKolczynski-NOAA replied:
But `gfs` and `gdas` already both call `FV3_GFS_nml`, why can't `gefs`?

XianwuXue-NOAA replied:
Good insight. I can delete FV3_GEFS_nml to use FV3_GFS_nml

WalterKolczynski-NOAA replied:
You got rid of this one, but there are still a few others that weren't removed.

XianwuXue-NOAA committed:
Add braces around variables to fix the lint warning

 On branch feature/gefs_v13_822_add_atm_fcst
 Changes to be committed:
\	modified:   ush/forecast_det.sh

Refs: #822

WalterKolczynski-NOAA commented on ush/forecast_postdet.sh:
```diff
@@ -599,14 +630,22 @@ data_out_GFS() {
       fi
-    elif [ $CDUMP = "gfs" ]; then
-      $NCP $DATA/input.nml $ROTDIR/${CDUMP}.${PDY}/${cyc}/atmos/
+    elif [ $CDUMP = "gfs" ] || [ ${RUN} = "gefs" ]; then
```
Instead of the nested if, use an additional `elif`.

XianwuXue-NOAA replied:
I think when the GFS new COM is ready, we don't need the "if [ ${RUN} = "gefs" ]; then" conditional. So this is only a temporary solution.

WalterKolczynski-NOAA replied:
Still think it should be a separate elif. We'll recombine later if necessary (really, I think copying the `input.nml` can just be done for all).

XianwuXue-NOAA replied:
agree that copying the input.nml is done for all. If separate it, I should be in the else, because the if conditional use $CDUMP, but for GEFS, it should use $RUN to identify it is for GEFS. What do you think?

WalterKolczynski-NOAA commented on ush/forecast_predet.sh:
```diff
@@ -241,13 +241,19 @@ FV3_GFS_predet(){
+  memdir=${memdir:-$ROTDIR/${prefix}.$PDY/$cyc/atmos/$memchar}
   if [ ! -d $memdir ]; then mkdir -p $memdir; fi
+  if [[ $RUN == "gefs" ]]; then
+    if [ ! -d ${memdir}/history ]; then mkdir -p ${memdir}/history; fi
```
With `mkdir -p`, you shouldn't need to check if the directory already exists first.

XianwuXue-NOAA replied:
I added the check to avoid run mkdir if it is already exists. Of course, it doesn't matter for "mkdir -p". I can delete the check if you want.

XianwuXue-NOAA commented:
@aerorahul @WalterKolczynski-NOAA Though there still have some warnings after the lint ShellCheck, I will not fix them because those warnings are not the modification in this PR. if you think I should fix the warnings, please let me know.

XianwuXue-NOAA committed:
Remove FV3_GEFS_nml and use FV3_GFS_nml for GEFS

 On branch feature/gefs_v13_822_add_atm_fcst
 Changes to be committed:
	modified:   scripts/exglobal_forecast.sh
	modified:   ush/forecast_postdet.sh

Refs: #822

XianwuXue-NOAA committed:
Remove the new COM dirs

 On branch feature/gefs_v13_822_add_atm_fcst
 Changes to be committed:
	modified:   ush/forecast_postdet.sh

Refs: #822

XianwuXue-NOAA commented on ush/forecast_postdet.sh:
```diff
@@ -599,14 +600,18 @@ data_out_GFS() {
-    elif [ $CDUMP = "gfs" ]; then
-      $NCP $DATA/input.nml $ROTDIR/${CDUMP}.${PDY}/${cyc}/atmos/
+    elif [ $CDUMP = "gfs" ] || [ ${RUN} = "gefs" ]; then
+      $NCP $DATA/input.nml ${ROTDIR}/${RUN}.${PDY}/$cyc/${mem:-""}/atmos/
```
In this change, I use ${RUN}.${PDY} instead of ${CDUMP}.${PDY} based on the NCO's standard. I also add the ${mem:-""} to deal with GEFS data.

XianwuXue-NOAA commented:
@WalterKolczynski-NOAA @aerorahul I just did the changes to remove the history and master folders. Please review my changes.

aerorahul commented on scripts/exgdas_enkf_fcst.sh:
```diff
@@ -108,7 +108,9 @@ export LEVS=${LEVS_ENKF:-${LEVS:-64}}
 # nggps_diag_nml
 export FHOUT=${FHOUT_ENKF:-3}
-
+if [[ $CDUMP == "gfs" ]]; then
+    export FHOUT=${FHOUT_ENKF_GFS:-${FHOUT_ENKF:${FHOUT:-3}}}
+fi
```
Instead, how about this on the line above?
```suggestion
export FHOUT=${FHOUT_ENKF_${CDUMP}:-${FHOUT_ENKF:-${FHOUT:-3}}}
```

XianwuXue-NOAA replied:
$CDUMP should be "gfs", not "GFS", should I change FHOUT_ENKF_GFS to FHOUT_ENKF_gfs?

aerorahul replied:
That makes sense.

XianwuXue-NOAA replied:
To be consistent, I think using FHOUT_ENKF_GFS is better than FHOUT_ENKF_gfs, because we don't have FHOUT_ENKF_GDAS variable. So I think I don't need to change it.

aerorahul commented on ush/forecast_predet.sh:
```diff
@@ -266,6 +272,14 @@ FV3_GFS_predet(){
 }
 
+FV3_GEFS_predet(){
+  echo "SUB ${FUNCNAME[0]}: Defining variables for FV3GFS"
```
```suggestion
  echo "SUB ${FUNCNAME[0]}: Defining variables for FV3GEFS"
```

XianwuXue-NOAA committed:
Correct a typo

 On branch feature/gefs_v13_822_add_atm_fcst
 Changes to be committed:
	modified:   ush/forecast_predet.sh

Refs: #822

XianwuXue-NOAA committed:
Merge branch 'feature/gefs_v13_822_add_atm_fcst' of github.com:XianwuXue-NOAA/global-workflow into feature/gefs_v13_822_add_atm_fcst

XianwuXue-NOAA committed:
Delete the unused statements

 On branch feature/gefs_v13_822_add_atm_fcst
 Changes to be committed:
	modified:   ush/forecast_predet.sh

Refs: #822

XianwuXue-NOAA committed:
Modify to save RESTART files for both gfs and gefs

 On branch feature/gefs_v13_822_add_atm_fcst
 Changes to be committed:
	modified:   ush/forecast_predet.sh

Refs: #822

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/gefs_v13_822_add_atm_fcst

 On branch feature/gefs_v13_822_add_atm_fcst

aerorahul commented on ush/forecast_det.sh:
```diff
@@ -41,7 +41,7 @@ FV3_GFS_det(){
   RERUN="NO"
   filecount=$(find "${RSTDIR_ATM:-/dev/null}" -type f | wc -l)
-  if [ ${CDUMP} = "gfs" -a ${rst_invt1} -gt 0 -a ${FHMAX} -gt ${rst_invt1} -a ${filecount} -gt 10 ]; then
+  if [[ ( ${CDUMP} = "gfs" || ( ${RUN} = "gefs" && ${CDATE_RST} = "" )) && ${rst_invt1} -gt 0 && ${FHMAX} -gt ${rst_invt1} && ${filecount} -gt 10 ]]; then
```
Is there a way to use `$RUN` or `$CDUMP` consistently.
I think replacing `$CDUMP` with `$RUN` would work here.
What is `$CDATE_RST`?

XianwuXue-NOAA replied:
Because CDUMP can only be gfs or gdas in g-w, so I use RUN to determine this is for GEFS based on the exglobal_forecast.sh. 

For $CDATE_RST, is for RERUN=YES. See https://github.com/NOAA-EMC/global-workflow/blob/1a7f0746a78b04cf9e92ef49aca40a81d3d91f42/scripts/exglobal_fcst_nemsfv3gfs.sh#L158. I will do more research on it.

XianwuXue-NOAA replied:
CDATE_RST is to set the restart CDATE, especially for the  forecast_d16_35. Then forecast_d16_35 can run restart from FHMAXH instead of from FHMAXH-$restart_interval.

aerorahul commented on workflow/rocoto/workflow_tasks.py:
```diff
@@ -1259,9 +1259,10 @@ def epos(self):
             if self.cdump == "gfs":
                 fhmax = epos['FHMAX_ENKF_GFS']
-            fhout = epos['FHOUT_ENKF']
+                fhout = epos['FHOUT_ENKF_GFS']
```
If you make it `FHOUT_ENKF_gfs` in `config.base`, be use to make the same change here.

XianwuXue-NOAA replied:
Thanks. As mentioned in the previous answer, I think it is better to use FHOUT_ENKF_GFS since there is no variable FHOUT_ENKF_gdas. And also consistent with other variables, such as FHMAX_ENKF_GFS. How do you think?

aerorahul reviewed (commented):
see some comments.

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/gefs_v13_822_add_atm_fcst

XianwuXue-NOAA committed:
Make RERUN receive value from out of the script

This is mostly for GEFS forecast_d16_35 job

 On branch feature/gefs_v13_822_add_atm_fcst
	modified:   ush/forecast_det.sh

Refs: #822

XianwuXue-NOAA committed:
Delete the GEFS functions to reuse GFS functions
 On branch feature/gefs_v13_822_add_atm_fcst
	modified:   scripts/exglobal_forecast.sh
	modified:   ush/forecast_det.sh
	modified:   ush/forecast_postdet.sh
	modified:   ush/forecast_predet.sh

Refs: #822

XianwuXue-NOAA commented:
@WalterKolczynski-NOAA @aerorahul I think I have modified the PR based on your comments. Please find time to review it. I think maybe I can add 'export rCDUMP="gdas"' to this PR to fix the bug if you agree.

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/gefs_v13_822_add_atm_fcst

WalterKolczynski-NOAA commented on ush/forecast_predet.sh:
```diff
@@ -208,8 +208,8 @@ FV3_GFS_predet(){
-  if [ $CDUMP = "gfs" ] && [ $rst_invt1 -gt 0 ] && [ $MEMBER -lt 0 ]; then
-    RSTDIR_ATM=${RSTDIR:-$ROTDIR}/${CDUMP}.${PDY}/${cyc}/atmos/RERUN_RESTART
+  if [[ $CDUMP = "gfs" || $RUN = "gefs" ]] && [ $rst_invt1 -gt 0 ]; then
+    RSTDIR_ATM=${RSTDIR_ATM:-${RSTDIR:-$ROTDIR}/${CDUMP}.${PDY}/${cyc}/atmos/RERUN_RESTART}
```
Do we need to have both `$RSTDIR` or `$RSTDIR_ATM`?

XianwuXue-NOAA replied:
I didn't find "RSTDIR=" in the g-w repo. However, I am not sure whether some others need it or not. So I keep it. If you think RSTDIR can be removed, then I will remove it.

WalterKolczynski-NOAA replied:
I don't see it either. Just remove it and only use RSTDIR_ATM

XianwuXue-NOAA replied:
Done

WalterKolczynski-NOAA commented on scripts/exgdas_enkf_fcst.sh:
```diff
@@ -108,7 +108,9 @@ export LEVS=${LEVS_ENKF:-${LEVS:-64}}
 # nggps_diag_nml
 export FHOUT=${FHOUT_ENKF:-3}
-
+if [[ $CDUMP == "gfs" ]]; then
```
Any new code should comply with shellcheck (always use `{ }`).

XianwuXue-NOAA replied:
Done

WalterKolczynski-NOAA commented on ush/forecast_predet.sh:
```diff
@@ -208,8 +208,8 @@ FV3_GFS_predet(){
   #-------------------------------------------------------
-  if [ $CDUMP = "gfs" ] && [ $rst_invt1 -gt 0 ] && [ $MEMBER -lt 0 ]; then
-    RSTDIR_ATM=${RSTDIR:-$ROTDIR}/${CDUMP}.${PDY}/${cyc}/atmos/RERUN_RESTART
+  if [[ $CDUMP = "gfs" || $RUN = "gefs" ]] && [ $rst_invt1 -gt 0 ]; then
```
More `${ }` needed

XianwuXue-NOAA replied:
Done

WalterKolczynski-NOAA commented on ush/forecast_predet.sh:
```diff
@@ -232,7 +232,7 @@ FV3_GFS_predet(){
   #-------------------------------------------------------
   # member directory
-  if [ $MEMBER -lt 0 ]; then
+  if [[ $MEMBER -lt 0 || $RUN = "gefs" ]]; then
```
And here

XianwuXue-NOAA replied:
Done

WalterKolczynski-NOAA commented on ush/forecast_predet.sh:
```diff
@@ -241,13 +241,13 @@ FV3_GFS_predet(){
     memchar=mem$(printf %03i $MEMBER)
   fi
-  memdir=$ROTDIR/${prefix}.$PDY/$cyc/$memchar/atmos
+  memdir=${memdir:-$ROTDIR/${prefix}.$PDY/$cyc/$memchar/atmos}
```
And here

XianwuXue-NOAA replied:
Done

WalterKolczynski-NOAA commented on ush/forecast_predet.sh:
```diff
@@ -241,13 +241,13 @@ FV3_GFS_predet(){
   gPDY=$(echo $GDATE | cut -c1-8)
   gcyc=$(echo $GDATE | cut -c9-10)
-  gmemdir=$ROTDIR/${rprefix}.$gPDY/$gcyc/$memchar/atmos
+  gmemdir=${gmemdir:-$ROTDIR/${rprefix}.$gPDY/$gcyc/$memchar/atmos}
```
And here

XianwuXue-NOAA replied:
Done.

XianwuXue-NOAA committed:
Add { } to variables

 On branch feature/gefs_v13_822_add_atm_fcst
 Changes to be committed:
	modified:   scripts/exgdas_enkf_fcst.sh
	modified:   ush/forecast_predet.sh

Refs: #822

XianwuXue-NOAA commented on scripts/exgdas_enkf_fcst.sh:
```diff
@@ -108,7 +108,9 @@ export LEVS=${LEVS_ENKF:-${LEVS:-64}}
 # nggps_diag_nml
 export FHOUT=${FHOUT_ENKF:-3}
-
+if [[ ${CDUMP} == "gfs" ]]; then
```
This checking is for early cycle enkf.

XianwuXue-NOAA commented:
@WalterKolczynski-NOAA I did the changes by add {}. Thanks

XianwuXue-NOAA commented:
@WalterKolczynski-NOAA @aerorahul Do you have any comments about this PR?

XianwuXue-NOAA committed:
Remove unused RSTDIR
 On branch feature/gefs_v13_822_add_atm_fcst
 Changes to be committed:
	modified:   ush/forecast_predet.sh

Refs: #822

WalterKolczynski-NOAA reviewed (approved):
Approved pending additional testing. Please confirm things still work for a cycled experiment using global-workflow.

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/gefs_v13_822_add_atm_fcst

XianwuXue-NOAA commented:
> Approved pending additional testing. Please confirm things still work for a cycled experiment using global-workflow.

I have tested 1.5 cycles of early cycle enkf and 1 cycle of GEFS workflow, everything works fine.

XianwuXue-NOAA committed:
Create links for final restart written files only
  when FHMAX_GFS%restart_interval_gfs=0

These changes will avoid to create invalid links for the final
  restart writting files

 On branch feature/gefs_v13_822_add_atm_fcst
 Changes to be committed:
	modified:   ush/forecast_predet.sh

Refs: #822

XianwuXue-NOAA commented:
@WalterKolczynski-NOAA @aerorahul I pushed one comment to avoid create invalid link for the final restart files.

XianwuXue-NOAA committed:
Change memdir/INPUT using ICSDIR to use dif ICSDIR

 On branch feature/gefs_v13_822_add_atm_fcst
	modified:   ush/forecast_postdet.sh

Refs: #822

XianwuXue-NOAA committed:
Remove unused function

 On branch feature/gefs_v13_822_add_atm_fcst
	modified:   ush/forecast_postdet.sh

Refs: #822

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/gefs_v13_822_add_atm_fcst

WalterKolczynski-NOAA commented on ush/forecast_postdet.sh:
```diff
@@ -134,7 +129,8 @@ EOF
 
   else ## cold start
-    for file in $(ls $memdir/INPUT/*.nc); do
+    ICSDIR=${ICSDIR:-${memdir}/INPUT}         # cold start initial conditions
```
I'm not sure if this is going to work or not. `config.base` sets `ICSDIR` with a value provided by `setup_exp.py`, but if one isn't provided, the value is `None`, to the fallback would never be used here. This will cause issues if people stage ICs manually. Plus, `setup_exp.py` should've already copied the ICs to the `INPUT` directory already.

XianwuXue-NOAA replied:
The changes is to allow users can use different ICSDIR, otherwise, we mush put the ICS into the ${memdir}/INPUT.  Or I can use **ICSIDR_ATM** to be consistent with **RSTDIR_ATM**. How do you think?

XianwuXue-NOAA replied:
If user does not provide the ICSDIR, then ICSDIR will use default value: ${memdir}/INPUT

XianwuXue-NOAA replied:
I tested the forecast in GEFS repo to set ICSDIR, and it works well.

XianwuXue-NOAA replied:
@aerorahul @WalterKolczynski-NOAA I just removed the ICSDIR.

XianwuXue-NOAA committed:
Remove the ICSDIR in the forecast_postdef.sh

 On branch feature/gefs_v13_822_add_atm_fcst
	modified:   ush/forecast_postdet.sh

Refs: #822

XianwuXue-NOAA committed:
Optimize case statements in the script

 On branch feature/gefs_v13_822_add_atm_fcst
	modified:   scripts/exglobal_forecast.sh

Refs: #822
